### ISSUE TITLE
Feature Request: Basic Markup for Rants

### ISSUE THREAD

**Issue by Elandor64** [devRant/devRant#27]
Maybe markdown-like. Especially a code markup would be cool.

**Comment by NicholasBatesNZ** [comment]
I agree that some level of markdown support would be beneficial however it does also introduce the issue of homework spam. While I as a devRant user would appreciate short snippets for ranting purposes, I do not want to have my feed filled with questions about how to fix some bug.
Perhaps something to be considered is instead having the user include a link to a gist/pastebin/etc. page and then embedding only the first 5(?) lines of said code into the rant to keep it as a sample. 

_Sent from my Google Nexus 5X using [FastHub](https://play.google.com/store/apps/details?id=com.fastaccess.github)_

**Comment by cozyplanes** [comment]
I am not really into this at all, but i would prefer the basics, like __bold__, _italics_, and `inline code`

But no headers, lists, strikeout, checkbox, divider, codeblocks, quotes please... I think those will make rants messier.

**Comment by Bear-with-me** [comment]
Those *extra* characters are **really** going to blow out character counts though. Might also need to pad the character limit with an extra 25-50 characters.

**Comment by DFoxinator** [comment]
I think this is an interesting issue that I've seen many opinions on. I'm in the boat that some limited markdown (pretty much only bold, italics, code) could be beneficial, but should also never show up on the main feed to prevent abuse/usage to simply try to promote more views of the rant.

The other significant issue is actual implementation. As far as I can tell, it really isn't straightforward. We'd have to provide a method via the rant API to tell different clients about how the rant is formatted. This is tricky I think, but could be done. I'm not sure how exactly is the best way to tell various clients about formatting. I believe one of the most common formatting techniques for mobile apps is attributed strings (and that's what Titanium uses), so maybe something like that, but not exactly sure.

Ideas are welcome!

**Comment by fox8091** [comment]
This is an old issue, but I feel I have a simple solution. Interpret every rant as Markdown. Plain text in Markdown is plain text, so no change for old rants, so, no need to tell them how it is formatted. As far as display, a limited Markdown interpreter should suffice. For the web portion, even a regex would work for some of the simpler stuff. For mobile however, I'm not quite sure.

**Comment by mannyhappenings** [comment]
Basic markdown I have been using on devRant until now despite for no-support (assuming fellow users are devs and understand markdown source),
- Links
- Bold
- Inline code (for names mainly)
- Full codes (small snippets 4-5 lines), well sometimes while replying

**Comment by xlf1024** [comment]
same opinion here, basic markdown is easily human-readable, I understood it even * before * I first learned about the existence of markdown. (Saw it being used on devRant and Twitter). Thus, I'd not worry too much about compatibility.

**Comment by cozyplanes** [comment]
Just two asterisk for **bold**
One asterisk (or one underscore) for _italic_ is sufficient

**Comment by jonasgeiler** [comment]
@cozyplanes Yeah like WhatsApp

**Comment by cprn** [comment]
How do people paste in blocks of text with new lines that isn't automatically interpreted as paragraphs? It seems like some kind of formatting already, e.g. [1st 3 lines of this post](https://devrant.com/rants/2412493/root-struggles-with-her-ticket-boss-struggles-too-also-random-thoughts-about-thi):

```
> Root struggles with her ticket
> Boss struggles too
> Also: random thoughts about this job
```

When I copy and paste this into a comment it renders like so:

```
> Root struggles with her ticket

> Boss struggles too

> Also: random thoughts about this job
```

And when I edit that comment I see there were extra line feeds added. People somehow prevent that. Some special characters? Maybe similar special characters can be used for more formatting?
